mlflow/recipes/steps/train.py

```diff
@@ -418,18 +470,25 @@ def my_warn(*args, **kwargs):
                 }
             )
             train_predictions = model.predict(raw_train_df.drop(self.target_col, axis=1))
-            if self.predict_proba:
-                train_predicted_probs = model.predict_proba(
-                    raw_train_df.drop(self.target_col, axis=1)
+            if isinstance(train_predictions, pd.DataFrame) and {
+                f"{self.predict_prefix}label",
+                f"{self.predict_prefix}score",
+            }.issubset(train_predictions.columns):
+                train_predicted_result = model.predict(raw_train_df.drop(self.target_col, axis=1))
+                train_predicted_probs = train_predicted_result.drop(
+                    [f"{self.predict_prefix}label", f"{self.predict_prefix}score"], axis=1
                 )
                 predicted_training_data = raw_train_df.assign(
-                    predicted_data=train_predictions,
-                    predicted_probability=train_predicted_probs[:, 0],
+                    predicted_data=train_predictions[f"{self.predict_prefix}label"],
+                    predicted_probability=train_predicted_probs.iloc[0:, 0].values,
                 )
                 worst_examples_df = BaseStep._generate_worst_examples_dataframe(
                     raw_train_df,
-                    train_predicted_probs[:, 0],
-                    error_fn(train_predicted_probs, raw_train_df[self.target_col].to_numpy()),
+                    train_predicted_probs.iloc[0:, 0].values,
+                    error_fn(
+                        train_predicted_probs.iloc[0:].values,
```

Comment:
Would this work instead?

```suggestion
                        train_predicted_probs.values,
```